### DISCUSSION TITLE
Reorder fields items

### DISCUSSION THREAD

**Discussion by und3fined** [ory/kratos#1196] [Q&A] [answered]
Hi team,

Currently, i'm make a service ID for my project. And with render register form is bad ux

<img width="502" alt="Screen Shot 2021-04-01 at 11 43 11 AM" src="https://user-images.githubusercontent.com/2163878/113244950-3259af00-92e0-11eb-94e2-3cf6074e3b0f.png">

Why password as first Input? How to make it show after email?

**Comment by aeneasr** [comment, marked as answer]
You can reorder it in your code.

**Reply by und3fined** [reply]
It's hard :)
I want to make a User Manager service, base from Kratos.
With your solution i need create case-by-case for my customer

It's not possible

**Reply by vinckr** [reply]
Can you share your frontend maybe @und3fined ? 
Ory Kratos comes headless, so all UI/UX matters you would have to implement yourself. 
We do have a reference implementation, you can check that out for some pointers:
https://github.com/ory/kratos-selfservice-ui-node

Does that make sense to you? 
Your use case is definitely possible with Ory Kratos.

**Reply by und3fined** [reply]
Currently, it's in development
But will avaliable in `id.und3fined.com`

**Reply by bryanveloso** [reply]
I ran into the same issue myself and looked at the node implementation. Unless I'm missing something, given that the node implementation (and the example in the docs) just iterates through the fields ([https://github.com/ory/kratos-selfservice-ui-node/blob/master/views/partials/form.hbs](https://github.com/ory/kratos-selfservice-ui-node/blob/master/views/partials/form.hbs)), the expected output would be the OP's screenshot.

**Reply by aeneasr** [reply]
With Ory Kratos 0.6 we improve the order you get out of Ory Kratos' APIs. If you need other ordering you can reorder the array in your code base using some sort function!

**Reply by zn8nz** [reply]
Somebody solved this problem already: [https://github.com/ellioseven/kratos-ss-ui-react/blob/master/src/components/KratosForm.tsx](https://github.com/ellioseven/kratos-ss-ui-react/blob/master/src/components/KratosForm.tsx)
